ui/src/data-services/models/occurrence.ts

```diff
@@ -38,6 +38,22 @@ export class Occurrence {
     return `${this._occurrence.determination.id}`
   }
 
+  get determinationIdentificationId(): string | undefined {
```

Comment:
Not super happy with this, I think a separate model representation for identifications/predictions could simplify, but perhaps not in this PR

Reply:
Okay yes. I'm curious to know more about why you need to make additional prop functions like `determinationPredictionId()` instead of accessing `determination_details` directly

Reply:
I want to keep the interface separated from BE as far as possible. That is why I have setup a separate layer to handle the all backend communication, including converting data. This means, when you are making an API update (like changing the structure of data), I don't have make updates all over the FE app. Instead I can just update the FE model class. 

Also I'm using this layer to simplify a bit. Sometimes I'm getting more than needed from BE and somtimes in a format that requires a bit of proccessing before rendering (I think taxa is one good example of this). In this way, I can choose what to expose to the interface layer and how.

I do think the classes are sometimes a bit overkill and there are definitely room for improvements here and there (like in code above), but some kind of FE representation of the data is pretty nice for me to have :) Happy to discuss more, maybe you see things from a different perspective?

Reply:
Thank you for that explanation! I really appreciate you keeping the frontend working while the backend has lagged behind. It makes sense to keep the objects in the frontend specific and simpler when they can be. But I also like the idea of eventually making types/models more consistent between the frontend & backend when the API is stable. At some point would even like to try using the types generated by the OpenAPI SDK directly in React. We can have multiple "view" types for a model (`OccurrenceIdentificationAgreeDetails`), but it would be awesome to define them in one place. Especially for validation, and for when we are all full stack developers!

Reply:
I would love that, it would make the FE app less fragile also, if we can use a more type safe approach!